extensions/cornerstone/src/commandsModule.ts

```diff
@@ -72,7 +96,129 @@ const commandsModule = ({ servicesManager }) => {
     return toolGroup;
   }
 
+  function _showViewerContextMenu(viewerElement, options) {
+    let defaultPointsPosition = [];
+    if (options.nearbyToolData) {
+      defaultPointsPosition = commandsManager.runCommand(
+        'getToolDataActiveCanvasPoints',
+        { toolData: options.nearbyToolData }
+      );
+    }
+
+    contextMenuController.showContextMenu(
+      options,
+      viewerElement,
+      defaultPointsPosition
+    );
+  }
+
   const actions = {
+    /** Show the specified context menu */
+    showViewerContextMenu: providedOptions => {
+      const viewerElement = _getActiveEnabledElement();
+
+      const options = { ...providedOptions };
+      const { event: evt } = options;
+      const { useSelectedAnnotation, nearbyToolData, menuName } = options;
+
+      if (menuName) {
+        Object.assign(
+          options,
+          customizationService.getModeCustomization(
+            menuName,
+            defaultContextMenu
+          )
```

Comment:
I hate this pattern that you should pass the default inside. Compared to

```
customizationService.getModeCustomization(menuName) || defaultContextMenu
```

can we fix this?

Reply:
No, the reason is that the default may need expansion inside to reference other registered GUI elements base on the uiType field.

Reply:
But, for the cases that you CAN do that, we could change the pattern.  This case happens to be one of them.